### PR TITLE
nrf_modem: v2.4.0

### PR DESCRIPTION
**Pull request by lemrey**
See CHANGELOG file for details.

### CODE REVIEW

**Review by eivindj-nordic — COMMENTED** [review]
Just some nits, will test when sdk-zepyr PR is added.

**Comment by lemrey** [comment]
Squashed and rebased.

**Review by divipillai — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: nrf_modem/doc/CHANGELOG.rst [hunk 1]
```diff
@@ -9,6 +9,47 @@ Changelog
...
+  * The :c:struct:`nrf_modem_at_cmd_filter` struct is renamed to :c:struct:`nrf_modem_at_cmd_custom`.
+  * The :c:func:`nrf_modem_at_cmd_filter_set` function is renamed to :c:func:`nrf_modem_at_cmd_custom_set`.
+
+* The ``paused`` field was removed from the :c:struct:`nrf_modem_at_cmd_custom`.
+  It is no longer possible to pause the dispatching of custom AT commands to their handler function.
+
+:ref:`Delta DFU <nrf_modem_delta_dfu>`
+======================================
+
+* It is no longer necessary to call the :c:func:`nrf_modem_shutdown` function after a updating the modem firmware.
```
Comment (eivindj-nordic) [inline nrf_modem/doc/CHANGELOG.rst:45]: nit:
```suggestion
* It is no longer necessary to call the :c:func:`nrf_modem_shutdown` function after updating the modem firmware.
```

**Review by eivindj-nordic — APPROVED** [review]
Modem library build version is correct.

**Pull request merged**